File: internal/port/capture.go
package port

import "ghostwire/internal/domain/screen"

type CaptureOptions struct {
	MaxWidth int
}

type ScreenCapturerPort interface {
	Start(opts CaptureOptions) error
	Bounds() screen.ScreenSize
	Grab() screen.ScreenFrame
	Close() error
}

File: src/core/ports/outbound/capture.ts
import type { ScreenFrame, ScreenSize } from '../../domain/screen/frame.js';

export interface CaptureOptions {
  maxWidth: number;
}

export interface ScreenCapturerPort {
  start(options: CaptureOptions): Promise<void>;
  bounds(): ScreenSize;
  grab(): ScreenFrame;
  close(): Promise<void>;
}
